### DISCUSSION TITLE
Element Connections Mappings not Showing up in the CP

### DISCUSSION THREAD

**Discussion by AbbeyDesign** [solspace/craft-freeform#901] [Questions & Answers] [answered]
I noticed in the cp that some of my forms' **Element Connections** are not rendering properly. These are forms with multiple fields pointing to a connected entry. The screenshot shows that now only **two** field mappings available. This form was set up with multiple fields mapped. Submitting forms functions properly. Mapping data to entries correctly, as they were initially set up but there's nothing in the cp. I need to add some fields but there is nothing available. 

<img width="311" alt="CleanShot 2023-10-05 at 22 03 43@2x" src="https://github.com/solspace/craft-freeform/assets/11862097/28d3f8eb-d9e5-444c-926b-7a0382c7fbed">

This is not happening on all the forms I have set up. Some show the correct mappings in the cp.

- Craft Version: 4.5.6.1
- Freeform Version: 4.1.9
- Freeform Edition: Pro
- Upgrade:

**Additional context**

**Comment by kjmartens** [comment]
Hi @AbbeyDesign,

I'm very sorry that this is happening to you. Can you answer a few questions to clarify the situation? 🙂 

1. Did this seem to happen after a recent upgrade of Freeform? If so, do you know which version you used last?
2. Did anything change in the Craft Entry sections' fields for the affected forms? For example, on form A, was a field removed or modified in the section X that form A is mapping to? (This shouldn't cause an issue, but maybe there's a bug related here).
3. If you try deleting the broken Element mapping inside the form builder (clicking the X at the top right) and then save the form and reload the browser, does it then show all the fields for mapping when you attempt to add it back?

**Comment by AbbeyDesign** [comment]
> Did this seem to happen after a recent upgrade of Freeform? If so, do you know which version you used last?

I update often and recently just noticed this. I can't attribute it to a specific update.

>Did anything change in the Craft Entry sections' fields for the affected forms?

Not sure I made changes , but all the forms that seem to have this bug point to one specific section. All the other forms pointing to other sections work properly

>...deleting the broken...

Tried this and it did not fix the issue... I am left with the same two fields ot connect

Again, the forms function/map properly when filled out, its just in the cp.

**Comment by AbbeyDesign** [comment]
This could have happened a long, long time ago. It's just now I noticed because I needed to make an adjustment to the mappings.

**Comment by kjmartens** [comment]
Sorry for the delay, @AbbeyDesign.

1. If you create a brand new test form, does it observe the same behavior?
2. Are you seeing any JS errors in the browser console while attempting to view/edit/add the connection?
3. Could you copy the JSON from the `layoutJson` column of an affected (and preferably simpler/smaller) form in the `freeform_forms` table and paste it here? If you'd rather not share it publicly, you can also send it in a private support ticket. 🙂

**Comment by AbbeyDesign** [comment]
I appologize for the delay in responding. I might have found where the issue is originating.

Upon further investigation the issue is present where the element connection is with any channel where its entry type has an auto-generated title. I changed the channel's title field to "Show the Title field" and the connection showed up correctly.

I don't know if the problem is the choice of auto generated title field or the fact that changing this initiated a resave entries queue and the entries just needed to be resaved. There are tens of thousands and I have to wait for the queue to run.

**Comment by AbbeyDesign** [comment]
It certainly has someting to do with the auto-generated titles. Just resaving the entries without changing the title functionality left the problem unchanged. I changed the channel to no longer auto generate the titles and the problem went away.

**Comment by kjmartens** [comment, marked as answer]
Sorry for the late response @AbbeyDesign. I revisited this one now, and I was able to duplicate the weird limitation of fields from some sections, but not all of them. What I noticed was that once I resaved the entry type (with field layout) in the section itself, it then populated the fields correctly for the Element Connection feature.

It seems that the resaving process resolves it, as I switched a channel that had auto-generated titles to now have Title field, saved, and it displays correctly in Element Connections. However, after I switched it back to auto-generated titles, it continued to work. I also observed this behavior in a section that didn't use this auto-generated titles feature at all. 🙂
